### PR TITLE
Mixed Precision batchnorm fix

### PR DESCRIPTION
**Pull request by rohan-varma**
Rehash of https://github.com/pytorch/pytorch/pull/76642 which could not be updated due to GHF out of sync issue.

### CODE REVIEW

**Review by awgu — APPROVED** [review]

**Comment by rohan-varma** [comment]
CI failure is unrelated - https://github.com/pytorch/pytorch/runs/6376377904?check_suite_focus=true

**Comment by suo** [comment]
@pytorchbot revert this 

Broke master: https://hud.pytorch.org/pytorch/pytorch/commit/bf61b795031b4f30b2cf1267c5625bfe36cd5f3c

**Comment by rohan-varma** [comment]
oh, the policy was renamed to: size_based_auto_wrap_policy
